rsf/sakai-rsf-core/src/java/org/sakaiproject/rsf/template/SakaiBodyTPI.java

```diff
@@ -33,17 +33,29 @@
 
 import java.util.Map;
 
+import org.sakaiproject.portal.util.PortalUtils;
 import uk.org.ponder.rsf.content.ContentTypeInfoRegistry;
 import uk.org.ponder.rsf.template.ContentTypedTPI;
 import uk.org.ponder.rsf.template.XMLLump;
 
 public class SakaiBodyTPI implements ContentTypedTPI {
 	public static final String SAKAI_BODY = "sakai-body";
+	public static final String SCRIPT = "script";
+	public static final String SRC = "src";
+	public static final String LINK = "link";
+	public static final String HREF = "href";
+	public static final String VERSION = "version=";
 
 	public void adjustAttributes(String tag, Map attributes) {
 		if (tag.equals("body") && attributes.get(XMLLump.ID_ATTRIBUTE) == null) {
 			attributes.put(XMLLump.ID_ATTRIBUTE, XMLLump.SCR_PREFIX + SAKAI_BODY);
 		}
+		else if (tag.equals(SCRIPT) && attributes.get(SRC)!=null && !((String)attributes.get(SRC)).contains(VERSION)) {
```

Comment:
Better check is to see if it contains any query parameters as the `getCDNQuery()` could append `?expire=1234`. Also if the URL already contains query parameters then it will make the URL bad (question marks).